addons/web_editor/static/src/scss/wysiwyg.scss

```diff
@@ -28,110 +40,240 @@ $o-we-zindex: $o-we-overlay-zindex + 1 !default;
-    .dropdown-toggle::after {
-        display: none;
-    }
-    .dropdown-menu.show { // To overcome .note-XXX .dropdown-menu rules
-        min-width: 0;
-    }
+
     .dropdown-item { // To overcome wysiwyg rules breaking this in iframes
-        display: block;
-        max-width: none;
-        overflow: visible;
-        margin-top: 0;
-        padding: 0 1em;
-        border: none;
-        background: none;
-        background-clip: padding-box;
-        background-color: $o-we-dropdown-item-bg;
-        color: $o-we-dropdown-item-color;
-        line-height: $o-we-dropdown-item-height;
+        display: flex;
+        align-items: center;
+        justify-content: flex-start;
+        padding: 0 $o-we-dropdown-item-height * .8;
+        min-height: $o-we-dropdown-item-height;
 
-        &:not(.d-none) ~ .dropdown-item {
-            // Use a border-top instead of a margin-top as when the
-            // mouse goes from one select button to another, the
-            // option preview should switch from the first button's
-            // option to the second one without reset to selected
-            // state in between.
-            border-top: $o-we-dropdown-item-spacing solid transparent;
+        > * {
+            color: inherit;
         }
 
         &.active {
-            color: $o-we-dropdown-item-active-color;
+            &, &:hover, &:focus {
+                color: $o-we-bg-darker;
+
+                &:before {
+                    top: 0;
+                    transform: translate($o-we-dropdown-item-height * -.5, 0);
+                    line-height: $o-we-dropdown-item-height;
+                }
+            }
         }
     }
-    li > .dropdown-item {
-        border-top: $o-we-dropdown-item-spacing solid transparent;
-    }
 
     // Specific elements
-    #colorInputButtonGroup label:last-of-type .btn {
-        margin: 0 1px 0 -1px;
+    #decoration {
+        #removeFormat {
+            display: none;
+        }
+        .active ~ #removeFormat {
+            display: flex;
+        }
     }
-    .oe-tablepicker {
-        margin: 1px 2px -8px 2px !important;
 
-        .oe-tablepicker-cell {
-            width: 5px !important;
+    .colorpicker-group {
+        .dropdown-toggle::after {
+            display: none;
+        }
+
+        .colorpicker {
+            --o-colorpicker-bg-rgb: #{red($oe-bg)}, #{green($oe-bg)}, #{blue($oe-bg)};
+            --o-colorpicker-text-rgb: #{red($oe-color-text)}, #{green($oe-color-text)}, #{blue($oe-color-text)};
+
+            display: flex;
+            flex-direction: column-reverse;
+        }
+
+        .colorpicker-menu {
+            bottom: 100%;
+
+            .o_we_color_btn {
+                border-color: $oe-bg;
+                box-shadow: inset 0 0 0 1px rgba(#000, .5);
+            }
+
+            .o_colorpicker_reset {
+                border: 0;
+                background: $o-brand-primary;
+                color: color-yiq($o-brand-primary);
+            }
```

Comment:
I am really scare about all this stuff, why do we need so much override for the toolbar only? I mean: the design is good but it does not seem different from the rest of the UI so why so much CSS just for it?

Reply:
> it does not seem much different from the rest of the UI so why so much CSS just for it?

Actually it's very different... long story short, the template uses default bootstrap classes (eg `.btn`) and it can't  change 'cause this lib should work standalone in a separate repo*.

In Odoo the design of these classes is different across community/enterprise and no one of these two designs matches the desired result (that should stay consistent regardless by the version btw).

The only solution we have (had*) is to override. But the approach was the same even before, now we have more refinement code + more empty lines + variables declaration ...

* it actually changes now. I was not aware but apparently the odoo-editor lib has been included in odoo (https://github.com/odoo/odoo/commit/4600086e7a2831664cc104a143e1014870874427). We should be able to define the right design once, rather than having to override it.